tests/components/landisgyr_heat_meter/test_config_flow.py

```diff
@@ -0,0 +1,227 @@
+        return_value=True,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"device": "/dev/ttyUSB0"}
+        )
+
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "setup_serial_manual_path"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+@patch("homeassistant.components.landisgyr_heat_meter.config_flow.HeatMeterService")
+@patch("serial.tools.list_ports.comports", return_value=[mock_serial_port()])
+async def test_list_entry_fail(mock_port, mock_heat_meter, hass: HomeAssistant) -> None:
+    """Test select from list entry fails."""
+
+    mock_heat_meter().read.side_effect = Exception
+    port = mock_serial_port()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"device": port.device}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+@patch("homeassistant.components.landisgyr_heat_meter.config_flow.HeatMeterService")
+@patch("serial.tools.list_ports.comports", return_value=[mock_serial_port()])
+async def test_get_serial_by_id_realpath(
+    mock_port, mock_heat_meter, hass: HomeAssistant
+) -> None:
+    """Test getting the serial path name."""
+
+    mock_heat_meter().read.return_value = MockUltraheatRead("LUGCUH50", "123456789")
+    port = mock_serial_port()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    scandir = [MagicMock(), MagicMock()]
+    scandir[0].path = "/dev/ttyUSB1234"
+    scandir[0].is_symlink.return_value = True
+    scandir[1].path = "/dev/ttyUSB5678"
+    scandir[1].is_symlink.return_value = True
+
+    with patch("os.path") as path:
+        with patch("os.scandir", return_value=scandir):
+            path.isdir.return_value = True
+            path.realpath.side_effect = ["/dev/ttyUSB1234", "/dev/ttyUSB5678"]
+            result = await hass.config_entries.flow.async_configure(
+                result["flow_id"], {"device": port.device}
+            )
+    assert result["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result["title"] == "LUGCUH50"
+    assert result["data"] == {
+        "device": port.device,
+        "model": "LUGCUH50",
+        "device_number": "123456789",
+    }
+
+
+@patch("homeassistant.components.landisgyr_heat_meter.config_flow.HeatMeterService")
+@patch("serial.tools.list_ports.comports", return_value=[mock_serial_port()])
+async def test_get_serial_by_id_dev_path(
```

Comment:
We're missing a test for already_configured abort reason. Add a MockConfigEntry and start the flow.